examples/latin-shift-cipher-example.rs

```diff
@@ -4,14 +4,11 @@
 //! Cipher.
 use fiddler::{CipherText, Key, Message};
 use rand::thread_rng;
-use std::{error::Error, io, process, str::FromStr};
+use std::{error::Error, io, str::FromStr};
 
-fn main() {
+fn main() -> Result<(), Box<dyn Error>> {
     println!("\nWelcome to the Latin Shift Cipher Demo!");
-    if let Err(e) = menu() {
-        eprintln!("Application error: {e}");
-        process::exit(1);
-    }
+    menu().inspect_err(|e| eprintln!("Application error: {e}"))
```

Comment:
I'm not sure I understand this change. What gets passed as the exit code to the underlying OS if there is an error?

Reply:
If there is an error, `main` here would return the canonical failure exit code for the platform (which is usually 1 on Unix). [Rust Playground Example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021. This section of [The Book](https://doc.rust-lang.org/book/ch09-02-recoverable-errors-with-result.html#where-the--operator-can-be-used) (towards the end) shows and  explains this pattern (see also the [ExitCode](https://doc.rust-lang.org/std/process/struct.ExitCode.html). 

I should have probably included that in my PR description, apologies for leaving this unclear!